src/sync.ts

```diff
@@ -562,7 +563,11 @@ export class SyncApi {
     };
 
     private buildDefaultFilter = () => {
-        return new Filter(this.client.credentials.userId);
+        const filter = new Filter(this.client.credentials.userId);
+        if (this.client.canSupport.get(Feature.ThreadUnreadNotifications)) {
+            filter.setUnreadThreadNotifications(true);
```

Comment:
`setUnreadThreadNotifications` only uses the stable non-prefixed version but `this.client.canSupport.get(Feature.ThreadUnreadNotifications)` will return true if unstable is supported.

Reply:
Good catch! I've changed the code to use `UnstableValue`. 
I believe I had not done that in the first place after seeing how `setProp` works... it essentially treats `.` in property names as a nested level

Reply:
Maybe we should update `setProp` to take a variadic list of keys rather than a string to avoid this problem